src/rhsmlib/facts/host_collector.py

```diff
@@ -0,0 +1,95 @@
+
+import os
+
+import rhsm.config
+
+from rhsmlib.facts import hwprobe
+from rhsmlib.facts import cleanup
+from rhsmlib.facts import custom
+from rhsmlib.facts import virt
+from rhsmlib.facts import firmware_info
+
+
+from rhsmlib.facts import collector
+
+
+class HostCollector(collector.FactsCollector):
+    """Collect facts for a host system.
+
+    'host' in this case means approx something running
+    a single kernel image. ie, regular x86_64 hardware, a KVM
+    virt guest, a ppc64 lpar guest. And not a cluster, or
+    a container, or an installroot/chroot/mock, or an application,
+    or a data center, or a distributed computing framework, or
+    a non-linux hypervisor, etc.
+
+    This in turns runs:
+        hwprobe.HardwareCollector()      [regular hardware facts]
+        virt.VirtCollector()    [virt facts, results from virt-what etc]
+        firmware_info.FirmwareCollector()  [dmiinfo, devicetree, etc]
+        cleanup.CleanupCollector()  [Collapse redundant facts, alter any
+                                     facts that depend on output of other facts, etc]
+
+
+    Facts collected include DMI info and virt status and virt.uuid."""
+
+    facts_sub_dir = 'facts'
+    facts_glob = '*.facts'
+
+    def get_all(self):
+        host_facts = {}
+        hardware_collector = hwprobe.HardwareCollector(
+            prefix=self.prefix,
+            testing=self.testing,
+            collected_hw_info=self._collected_hw_info
+        )
+        hardware_info = hardware_collector.get_all()
+
+        virt_collector = virt.VirtCollector(
+            prefix=self.prefix,
+            testing=self.testing,
+            collected_hw_info=self._collected_hw_info
+        )
+
+        virt_collector_info = virt_collector.get_all()
+
+        # AdminFacts includes VirtCollector and firmware facts collection
```

Comment:
I can't seem to find any other references to AdminFacts aside from this comment.